src/argowrapper/engine/helpers/argo_engine_helper.py

```diff
@@ -250,7 +250,7 @@ def convert_username_label_to_gen3username(label: str) -> str:
         return ""
 
 
-def _convert_to_label(special_character_match: str) -> str:
+def _convert_to_label(special_character_match: str) -> Optional[str]:
```

Comment:
does `-> Optional[str]:` mean that the function sometimes returns something and sometimes it returns None? The implementation here seems to always return something, so maybe this `-> Optional[str]:` is wrong?

Reply:
[`typing.Optional`](https://docs.python.org/3/library/typing.html#typing.Optional) is equivalent to `X | None` (or `Union[X, None]`).

Pylance was complaining about that, so I changed it:
```
Function with declared return type "str" must return value on all code paths
  "None" is not assignable to "str"
```